### ISSUE TITLE
findDOMNode doesn't work with `this` as argument

### ISSUE THREAD

**Issue by NulledGravity** [facebook/react#13652]
I'm getting an `Uncaught Error: Unable to find node on an unmounted component.` error with this component. It is a strange behaviour as `handleClickEvent` is attached only when the component is already mounted.

The issue is with `findDOMNode(this)`. For some reason it won't work. Is this the desired behaviour or I am missing something?

I'm running everything with Storybook and styled.components.
React/ReactDOM versions are 16.4.0.

```
import React, { Component, Children } from 'react';
import PropTypes from 'prop-types';
import { findDOMNode } from 'react-dom';

export default class extends Component {
  static propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func,
  };

  componentDidMount() {
    document.addEventListener('click', this.handleClickEvent, { capture: true });
  };

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClickEvent, { capture: true });
  };

  handleClickEvent = (event) => {
    const el = findDOMNode(this);
    
    if (
      !this.props.onClick ||
      !el ||
      (event.target instanceof Node && el.contains(event.target))
    ) {
      return;
    }

    this.props.onClick(e);
  };

  render() {
    return this.props.children;
  };
}
```

**Comment by gaearon** [comment]
Can you create an isolated fiddle demonstrating this?

**Comment by Simek** [comment]
I was NOT able to reproduce this issue. All three logs that I defined appears correctly in console in order `this` (React class), then HTML node found by `findDOMNode` and at the end `click` event passed to children. 

Test case that I created do not produce any warnings or errors on `16.4.0` version or on latest `16.5.1` version: https://codesandbox.io/s/7klp1j38rx

Feel free to edit/fork my fiddle and correct it to deliver behaviour as mentioned in issue.
